src/MakePotCommand.php

```diff
@@ -381,34 +427,99 @@ protected function makepot() {
-			if ( ! $translation->hasExtractedComments() ) {
-				continue;
+			/** @var Translation $translation */
+
+			$references = $translation->getReferences();
+
+			// File headers don't have any file references.
+			$location   = $translation->hasReferences() ? '(' . implode( ':', array_shift( $references ) ) . ')' : '';
+
+			if ( $translation->hasExtractedComments() ) {
+				$comments = $translation->getExtractedComments();
+				$comments_count = count( $comments );
+
+				if ( $comments_count > 1 ) {
+					WP_CLI::warning( sprintf(
+						'The string "%1$s" has %2$d different translator comments. %3$s',
+						$translation->getOriginal(),
+						$comments_count,
+						$location
+					) );
+				}
 			}
 
-			$comments = $translation->getExtractedComments();
-			$comments_count = count( $comments );
+			$non_placeholder_content = trim( preg_replace( '`^([\'"])(.*)\1$`Ds', '$2', $translation->getOriginal() ) );
+			$non_placeholder_content = preg_replace( self::SPRINTF_PLACEHOLDER_REGEX, '', $non_placeholder_content );
 
-			if ( $comments_count > 1 ) {
+			if ( '' === $non_placeholder_content ) {
 				WP_CLI::warning( sprintf(
-					'The string "%1$s" has %2$d different translator comments.',
-					$translation->getOriginal(),
-					$comments_count
+					'Found string without translatable content. %s',
+					$location
 				) );
 			}
-		}
 
-		$result = PotGenerator::toFile( $this->translations, $this->destination );
+			// UnorderedPlaceholders: Check for multiple unordered placeholders.
+			$unordered_matches_count = preg_match_all( self::UNORDERED_SPRINTF_PLACEHOLDER_REGEX, $translation->getOriginal(), $unordered_matches );
+			$unordered_matches = $unordered_matches[0];
 
-		$translations_count = count( $this->translations );
+			if ( $unordered_matches_count >= 2 ) {
+				WP_CLI::warning( sprintf(
+					'Multiple placeholders should be ordered. %s',
+					$location
+				) );
+			}
 
-		if ( 1 === $translations_count ) {
-			WP_CLI::debug( sprintf( 'Extracted %d string', $translations_count ), 'make-pot' );
-		} else {
-			WP_CLI::debug( sprintf( 'Extracted %d strings', $translations_count ), 'make-pot' );
+			if ( $translation->hasPlural() ) {
+				preg_match_all( self::SPRINTF_PLACEHOLDER_REGEX, $translation->getOriginal(), $single_placeholders );
+				$single_placeholders = $single_placeholders[0];
+
+				preg_match_all( self::SPRINTF_PLACEHOLDER_REGEX, $translation->getPlural(), $plural_placeholders );
+				$plural_placeholders = $plural_placeholders[0];
+
+				// see https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#plurals
+				if ( count( $single_placeholders ) < \count( $plural_placeholders ) ) {
+					WP_CLI::warning( sprintf(
+						'Missing singular placeholder, needed for some languages. See https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#plurals %s',
+						$location
+					) );
+				} else {
+					sort( $single_placeholders );
```

Comment:
I'm probably missing something here, but I think it warrants a comment. The goal here is to verify the ordering of the placeholders. However, before comparing them, you sort them here. Would that put them into the same order no matter what?

Reply:
I don't know what I was thinking here. I copied this from WPCS, but somehow messed it up. The order should be irrelevant, but different placeholders are usually not ok.